app/src/main/java/com/example/puzzlepic/RetrofitClientInstance.kt

```diff
@@ -13,14 +14,20 @@ object RetrofitClientInstance {
     val retrofitInstance : Retrofit?
 
     get(){
-        //has this object been created?
-        if(retrofit == null){
-            //then create it
-            retrofit = retrofit2.Retrofit.Builder()
+        createRetrofitObject()
+        return retrofit
+    }
+
+    fun createRetrofitObject() =
+        when(retrofit) {
+            null -> RetrofitClientInstance.retrofit = retrofit2.Retrofit.Builder()
                 .baseUrl(BASE_URL)
                 .addConverterFactory(GsonConverterFactory.create())
                 .build()
+            else -> {
+                //retrofit object already exists
+                File("application.log").writeText("Tried to create new retrofit object, object already exists")
```

Comment:
Logging is a good idea.  Use Log.i() (or Log.e(), Log.d(), etc.) to log statements.